### PR TITLE
Add marker map

### PR DESCRIPTION
**Pull request by TheImpressiveDonut**

### CODE REVIEW

File: app/src/androidTest/java/com/github/onedirection/map/OnMapReadyIdlingResource.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    public OnMapReadyIdlingResource(MapFragment fragment) {
+        this.fragment = fragment;
+        try {
+            Field field = fragment.getClass().getDeclaredField("mapView");
+            field.setAccessible(true);
+            ((MapView) field.get(fragment)).getMapAsync(this);
+        } catch (Exception err) {
+            throw new RuntimeException(err);
+        }
+    }
```
Comment (Ef55) [inline app/src/androidTest/java/com/github/onedirection/map/OnMapReadyIdlingResource.java:32]: You should initialize all attributes in the constructor (even if it is to `null`).

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java [hunk 2]
```diff
@@ -1,44 +1,104 @@
...
+import java.util.List;
 
 public class MapFragment extends Fragment {
 
     private MapView mapView;
+    private MapboxMap mapboxMap;
+    private MarkerSymbolManager markerSymbolManager;
+    private final String SYMBOL_ID = "MARKER_MAP";
+    private final String BUNDLE_ID = "bundleT";
+    private final String MARKERS_ID = "markers_size";
```
Comment (Ef55) [inline app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java:34]: Should most likely be static.

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java [hunk 3]
```diff
@@ -1,44 +1,104 @@
...
+                return false;
+            });
+        });
         return view;
     }
 
     @Override
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
+        System.out.println(savedInstanceState);
```
Comment (Ef55) [inline app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java:65]: Leftover debug.
If you want to keep it:
```suggestion
        private finale static String LOGCAT_ID = "MapFragment";
        ...
        Log.d(LOGCAT_ID, savedInstanceState.toString());
```

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java [hunk 4]
```diff
@@ -1,44 +1,104 @@
...
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        mapView.onSaveInstanceState(outState);
+    }
 
+    private void initializeMarkerSymbolManager(@NonNull Style styleOnLoaded) {
+        Drawable marker = ContextCompat.getDrawable(getContext(), R.drawable.ic_marker_map);
+        styleOnLoaded.addImage(SYMBOL_ID, BitmapUtils.getBitmapFromDrawable(marker));
+        SymbolManager symbolManager = new SymbolManager(mapView, mapboxMap, styleOnLoaded);
+        markerSymbolManager = new MarkerSymbolManager(symbolManager, SYMBOL_ID);
```
Comment (Ef55) [inline app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java:102]: I would write
```suggestion
        this.markerSymbolManager = new MarkerSymbolManager(symbolManager, SYMBOL_ID);
```
just to be more explicit (but that's just really my way of doing it).

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/MarkerSymbolManager.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import com.mapbox.mapboxsdk.plugins.annotation.Symbol;
+import com.mapbox.mapboxsdk.plugins.annotation.SymbolManager;
+import com.mapbox.mapboxsdk.plugins.annotation.SymbolOptions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MarkerSymbolManager {
+
+    private final String SYMBOL_ID;
```
Comment (Ef55) [inline app/src/main/java/com/github/onedirection/navigation/fragment/map/MarkerSymbolManager.java:13]: It shouldn't use SCREAMING_SNAKE_CASE if it's not static.

**Review by Ef55 — CHANGES_REQUESTED** [review]
Just a few details to fix; overall it's good.

**Review by Sosol02 — APPROVED** [review]

File: app/src/androidTest/java/com/github/onedirection/map/OnMapReadyIdlingResource.java (hunk 1 above)
Comment (TheImpressiveDonut) [reply app/src/androidTest/java/com/github/onedirection/map/OnMapReadyIdlingResource.java:32]: Done

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java (hunk 3 above)
Comment (TheImpressiveDonut) [reply app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java:65]: Removed

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java (hunk 2 above)
Comment (TheImpressiveDonut) [reply app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java:34]: Done

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java (hunk 4 above)
Comment (TheImpressiveDonut) [reply app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java:102]: I agree

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/MarkerSymbolManager.java (hunk 5 above)
Comment (TheImpressiveDonut) [reply app/src/main/java/com/github/onedirection/navigation/fragment/map/MarkerSymbolManager.java:13]: Removed it and change the use by MapFragment.SYMBOL_ID

**Review by Ef55 — APPROVED** [review]

**Pull request merged**